crypto/crypto/src/x25519.rs

```diff
@@ -206,6 +206,12 @@ impl ExchangeKey for X25519StaticPrivateKey {
     }
 }
 
+impl PartialEq for X25519StaticPrivateKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.to_bytes() == other.to_bytes()
```

Comment:
this should be constant time. if an attacker control one end of the comparison they may be able to get the private key. check dalek's subtle.

Reply:
I don't disagree, but note that if we're doing a comparison of a private key with an untrusted input, we might have a bigger problem.

Reply:
Should we be evaluating the API or when the API is used.

Reply:
it's always easier to make an API misuse-resistant, so if you think using a constant time check is worth it I'd say go for it.